### PR TITLE
Handle the new paypal sandbox guest checkout page

### PR DESCRIPTION
**Pull request by michaelwmcnamara**
Post production tests are failing on Paypal payments.
This is because paypal have introduced a new guest checkout page.
Unfortunately, unlike Stripe, the functionality does not exist to mock the paypal process so we must continue running through the payment flow.

This change checks whether a guest checkout page has opened, and if it has it navigates to the express checkout page so that payment flow can begin.

This work was done in response to a retrospective action tracked in trello card:
https://trello.com/c/DQUOaoW7/1491-stub-paypal-in-end-to-end-tests

### CODE REVIEW

File: test/selenium/OneOffContributionsSpec.scala [hunk 1]
```diff
@@ -36,10 +38,17 @@ class OneOffContributionsSpec extends FeatureSpec with GivenWhenThen with Before
...
       landingPage.clickContributePayPalButton
 
       Then("they should be redirected to PayPal Checkout")
-      assert(payPalCheckout.hasLoaded)
+      assert(payPalCheckout.initialPageHasLoaded)
+      val url = webDriver.getCurrentUrl
+      if (url.contains(payPalCheckout.guestRegistrationUrlFragment)) {
+        val token = url.substring(url.indexOf("&token="), url.indexOf(payPalCheckout.guestRegistrationUrlFragment))
+        webDriver.navigate().to(payPalCheckout.loginUrlFragment + token)
+        assert(payPalCheckout.loginContainerHasLoaded)
```
Comment (michaelwmcnamara) [inline test/selenium/OneOffContributionsSpec.scala]: move 2nd assert to outside of if container

File: test/selenium/RecurringContributionsSpec.scala [hunk 2]
```diff
@@ -95,11 +95,19 @@ class RecurringContributionsSpec extends FeatureSpec with GivenWhenThen with Bef
       And("they select to pay with PayPal")
 
       When("they press the PayPal payment button")
+      payPalCheckout.addPaypalCookie()(landingPage)
```
Comment (michaelwmcnamara) [inline test/selenium/RecurringContributionsSpec.scala]: get rid of

File: test/selenium/RecurringContributionsSpec.scala [hunk 3]
```diff
@@ -95,11 +95,19 @@ class RecurringContributionsSpec extends FeatureSpec with GivenWhenThen with Bef
...
       recurringContributionForm.selectPayPalPayment
 
       Then("the PayPal Express Checkout mini-browser should display")
       payPalCheckout.switchToPayPalPopUp
-      assert(payPalCheckout.hasLoaded)
+      assert(payPalCheckout.initialPageHasLoaded)
+      val url = webDriver.getCurrentUrl
```
Comment (michaelwmcnamara) [inline test/selenium/RecurringContributionsSpec.scala]: This should be a helper method

File: test/selenium/util/PayPalCheckout.scala [hunk 4]
```diff
@@ -13,6 +15,24 @@ class PayPalCheckout(implicit val webDriver: WebDriver) extends Browser {
   val passwordInput = name("login_password")
   val agreeAndPay = id("confirmButtonTop")
   val paymentAmount = className("formatCurrency")
+  val guestRegistrationUrlFragment = "#/checkout/guest"
+  val loginUrlFragment = "https://www.sandbox.paypal.com/webapps/hermes?country.x=US&hermesLoginRedirect=xoon&locale.x=en_US"
+
+  def addPaypalCookie()(implicit landing: ContributionsLanding): Unit = {
```
Comment (michaelwmcnamara) [inline test/selenium/util/PayPalCheckout.scala]: get rid of

File: test/selenium/RecurringContributionsSpec.scala [hunk 5]
```diff
@@ -99,10 +99,12 @@ class RecurringContributionsSpec extends FeatureSpec with GivenWhenThen with Bef
 
       Then("the PayPal Express Checkout mini-browser should display")
       payPalCheckout.switchToPayPalPopUp
-      assert(payPalCheckout.hasLoaded)
+      assert(payPalCheckout.initialPageHasLoaded)
+      payPalCheckout.handleGuestRegistrationPage()
```
Comment (jacobwinch) [inline test/selenium/RecurringContributionsSpec.scala]: Much cleaner 👍

**Pull request merged**